init.go

```diff
@@ -33,84 +42,74 @@ func opensslInit(file string) error {
 			_OPENSSL_INIT_LOAD_CONFIG|
 			_OPENSSL_INIT_LOAD_CRYPTO_STRINGS,
 		nil); err != nil {
-		dlclose(handle)
+		close()
 		return err
 	}
 	osslHandle = handle
 	return nil
 }
 
-// loadFuncs loads and initialize the OpenSSL functions.
-// See shims.go for the complete list of supported functions.
-func loadOpenSSLFuncs(handle unsafe.Pointer) {
-	mkcgoLoad_(handle)
-	if vMajor == 1 {
-		mkcgoLoad_legacy_1(handle)
-		if vPatch == 1 {
-			mkcgoLoad_111(handle)
-		}
-	} else {
-		mkcgoLoad_111(handle)
-		mkcgoLoad_3(handle)
-	}
-}
-
 // initForCheckVersion loads and initialize only the
 // functions required in [CheckVersion].
 // It returns a function that must be called to release the resources.
 // The function leaves all the global variables in the same state as they were
 // before the call.
 func initForCheckVersion(file string) (func(), error) {
-	// This function can be called when the
 	prevMajor, prevMinor, prevPatch := vMajor, vMinor, vPatch
```

Comment:
Not sure it's worth doing just for factoring's sake, but for readability, might be nice to have a name describing what's going on here:

```go
	prevMajor, prevMinor, prevPatch := vMajor, vMinor, vPatch
	restoreVersion := func() {
		vMajor, vMinor, vPatch = prevMajor, prevMinor, prevPatch
	}
```

(Maybe context from the func comment is plenty.)